File: armstrong-number/armstrong-number.c
#include <stdio.h>
#include <math.h>

int main() {
    int num = 153; // Change this to test other numbers
    int original = num, sum = 0, digits = 0;

    int temp = num;
    while (temp > 0) {
        digits++;
        temp /= 10;
    }

    temp = num;
    while (temp > 0) {
        int digit = temp % 10;
        sum += pow(digit, digits);
        temp /= 10;
    }

    if (sum == original)
        printf("%d is an Armstrong number.\n", original);
    else
        printf("%d is NOT an Armstrong number.\n", original);

    return 0;
}

File: armstrong-number/ArmstrongNumber.kt
fun main() {
    val num = 153  // Change this to test other numbers
    var temp = num
    var digits = 0
    var sum = 0

    // Count digits
    while (temp > 0) {
        digits++
        temp /= 10
    }

    temp = num
    while (temp > 0) {
        val digit = temp % 10
        sum += Math.pow(digit.toDouble(), digits.toDouble()).toInt()
        temp /= 10
    }

    if (sum == num) {
        println("$num is an Armstrong number.")
    } else {
        println("$num is NOT an Armstrong number.")
    }
}
